### ISSUE TITLE
bug: Node freezes when using the programmatic API

### ISSUE THREAD

**Issue by Brainy0207** [simonhaenisch/md-to-pdf#146]
**Context:**

- version (`md-to-pdf -v`): 5.2.0
- platform (Unix, macOS, Windows): Windows
- node version: 19.2.0

**Describe the bug:**

I have a really simple script:
```js
const { mdToPdf } = require('md-to-pdf');

async function convertAsync() {
  const pdf = await mdToPdf({ content: '# Hello, World' }, {dest: 'output.pdf'});
  console.log(pdf.filename);
}

convertAsync();
```
If I run this script with node, it logs the filename but freezes after that. I have to press Ctrl+C to get it to return.

It seems that multiple chrome.exe processes are created in the background and node is waiting for them to exit. If I kill these processes with the task manager, node returns.

This seems to be related to #141.
The browser instance created in `generateOutput` is never closed for the programmatic API:
https://github.com/simonhaenisch/md-to-pdf/blob/9f464aa32c349200d4f9dcf1bffa23dcf9b0f492/src/lib/generate-output.ts#L36-L38

And because `closeBrowser` is not exported to the top level, there seems to be no way to close this browser instance programmaticaly.

**Comment by simonhaenisch** [comment]
Thanks a lot for the bug report and digging into it, yeah I must have missed that 🙈 

I'll see that I get it fixed soon.

**Comment by simonhaenisch** [comment]
Turns out fixing this is a lot harder than I thought, mainly because of how the library behaves now when running multiple programmatic `mdToPdf` calls simultaneously (which happens also when running the tests).

There's two options to fix this now:

- PR #147: Track the ongoing programmatic conversions and only close the browser once there are no more active ones. This way the browser instance can be shared even between parallel programmatic calls. However it's not a very robust solution that's likely going to break in some edge cases, and that will be hard to fix.
- PR #148: Fall back to the previous behavior of one browser instance per programmatic conversion (which of course is slower).

At the cost of performance, I think I'd go with the second solution, and rather fix the performance problem in a better way (I've been planning to change the programmatic API to accept an array input anyway).

**Comment by simonhaenisch** [comment]
@Brainy0207 should be fixed in `5.2.1`, if you could verify that? Thanks!

**Comment by Brainy0207** [comment]
I ran my test script, no freezing, so I guess it is fixed. Thanks for the quick response.
